### ISSUE TITLE
Invalid for custom dataset

### ISSUE THREAD

**Issue by TaoistPriestC** [bethgelab/foolbox#673]
There are some datasets in foolbox, but in the `samples` function, the parameter `batch_size`only allow up to `20`. So I wanted to customize my datasets to test the adversarial  attacks. However, I found that no matter how I adjusted `epsilon`, the accuracy hardly changed.

```Python
## LeNet Model, used to test the effect of an attack
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
        self.conv2_drop = nn.Dropout2d()
        self.fc1 = nn.Linear(320, 50)
        self.fc2 = nn.Linear(50, 10)

    def forward(self, x):
        x = F.relu(F.max_pool2d(self.conv1(x), 2))
        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
        x = x.view(-1, 320)
        x = F.relu(self.fc1(x))
        x = F.dropout(x, training=self.training)
        x = self.fc2(x)
        return F.log_softmax(x, dim=1)
 
## I used pytorch's dataset instead of the dataset provided in the foolbox
train_loader = torch.utils.data.DataLoader(
    datasets.MNIST('D:/data', train=True, download=True,
	transform=transforms.Compose([
            transforms.ToTensor(),
            ])),
    batch_size=10, 
    shuffle=True
)
test_loader = torch.utils.data.DataLoader(
    datasets.MNIST('D:/data', train=False, download=True, 
	transform=transforms.Compose([
            transforms.ToTensor(),
            ])),
    batch_size=10, 
    shuffle=True
)

images_t = test_loader.dataset.data.unsqueeze(1)
labels_t  = test_loader.dataset.targets
shuffle_idx = torch.randperm(10000)
images_t = images_t[shuffle_idx]
labels_t = labels_t[shuffle_idx]

## ...
def attack_on_diff(attack):
    accuracy = []
    epsilons = [0.00, 0.03, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30]
    for i,eps in tqdm(enumerate(epsilons)):
        raw, clipped, is_adv = attack(
            fmodel, 
            images_t[:5000,:,:,:].float().to(device),
            labels_t[:5000].to(device),
            epsilons = eps
        )
        acc = 1 - is_adv.float().mean(axis=-1)
        accuracy.append(acc.item())
    return epsilons,accuracy
``` 

```Python
# Fast Gradient Method (FGM)
attack_on_diff(fb.attacks.L2FastGradientAttack())
```

```Python
## The return-value of function attack_on_diff
([0.0, 0.03, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3], 
 [0.9575999975204468,
  0.9575999975204468,
  0.9574000239372253,
  0.9574000239372253,
  0.9571999907493591,
  0.9571999907493591,
  0.9571999907493591,
  0.9571999907493591])
```
It can be seen that the accuracy hardly changes significantly with epsilon. The expected result should be similar with the figure below. I want to know, is this a bug, or there are something wrong in my code?

![image](https://user-images.githubusercontent.com/48847492/159821707-669407ed-9b3f-4cd8-baee-836816099a5d.png)

**Comment by jangop** [comment]
It appears that you are bypassing the data loader. I believe that also means your preprocessing will be skipped.

Iterate over the data loader instead. ☺️

And you probably want batches smaller than 5000. And the loader can shuffle for you. And during test you don't need to shuffle.

**Comment by zimmerrol** [comment]
Please make sure that the data you're using is in the `[0, 1]` value range. As far as I know, accessing the MNIST data from torchvision will return values in `[0, 255]`.

**Comment by Changgun-Choi** [comment]
Using 'transforms.ToTensor()' already makes in the [0, 1] value range. 
So we don't need to normalize in this case? 

ex.  transform = transforms.Compose([
        transforms.Resize((224, 224)),
        transforms.ToTensor(),
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
    ])

**Comment by TaoistPriestC** [comment]
Thanks! This problem has been solved!
